### DISCUSSION TITLE
Can I merge multiple managers to produce one MR?

### DISCUSSION THREAD

**Discussion by syphernl** [renovatebot/renovate#8749] [Request Help] [answered]
We have quite a few Terraform repositories within our company. When Hashicorp releases a new version of Terraform this causes manual work for us, even though Renovate does some heavy lifting.

Renovate:
- Updates the `.gitlab-ci.yml` Docker reference to the new version
- Updates the `.terraform-version`  to the new version
- Does not (yet?) update the version reference in the `required_versions` section (I have created a feature request for that: https://github.com/renovatebot/renovate/issues/8746)

The problem that we are facing is that two _separate_ MR's are being created for this right now. One to update the `.gitlab-ci.yml` and one to update the `.terraform-version`. 
Can we configure Renovate to create a single MR with both changes? 

The documentation on grouping mostly talks about grouping based on package patterns, if I try to go that route with 

```json
  {
    packageNames: ['hashicorp/terraform'],
    groupName: 'terraform',
  },
```

it simply creates two PR's (even though there already were two with the exact same changes) with branch `renovate/terraform` and `renovate/docker-terraform`.

Any pointers in what I need to configure to have it create a single MR for both changes? Thanks!

**Comment by rarkins** [comment, marked as answer]
Use `packageRules` to do matching and then assign a `groupName`. What are the current branch names which you'd like to see combined into one?

**Reply by viceice** [reply]
he also needs to override `additionalBranchPrefix` because docker datasource os overriding it by default

https://github.com/renovatebot/renovate/blob/01e1f38019f65bc4dfe29628870b5c0f2f954821/lib/datasource/docker/index.ts#L27

**Reply by syphernl** [reply]
I've now changed it to this:

```
  // Merge MR's updating hashicorp/terraform
  {
    packageNames: ['hashicorp/terraform'],
    groupName: 'terraform',
    additionalBranchPrefix: '',
  },
```

**Comment by viceice** [comment]
regex manager i use to update terraform `required_version` 😉 

https://github.com/visualon/renovate-config/blob/main/shared.json#L70-L76

```json
  {
      "fileMatch": ["\\.tf$"],
      "matchStrings": ["required_version\\s*=\\s*\"(?<currentValue>.+?)\""],
      "datasourceTemplate": "docker",
      "versioningTemplate": "hashicorp",
      "depNameTemplate": "hashicorp/terraform"
    }
```

**Reply by syphernl** [reply]
Ah that's cool! Thanks for sharing 👍🏻
